test/test_nn.py

```diff
@@ -7593,12 +7593,22 @@ def test_pointwise_loss_broadcast(self):
             'smooth_l1_loss': lambda x, y, r: F.smooth_l1_loss(x, y, reduction=r),
         }
 
+        supports_complex = ['l1_loss']
+
         input = torch.randn(2, 1, requires_grad=True)
-        for _name, fn in losses.items():
+        cinput = torch.randn(2, 1, requires_grad=True, dtype=torch.cfloat)
+        for name, fn in losses.items():
             for requires_grad in [True, False]:
                 # When target.requires_grad=True, its impl is in Python, while the other is in TH.
                 target = torch.randn(2, 10, requires_grad=requires_grad)
+                ctarget = torch.randn(2, 10, requires_grad=requires_grad, dtype=torch.cfloat)
                 for reduction in ['none', 'mean', 'sum']:
+                    if name in supports_complex:
+                        l = fn(cinput, ctarget, reduction)
+                        if reduction == 'none':
+                            self.assertEqual(l.size(), target.size())
+                        self.assertTrue(gradcheck(fn, (input, target, reduction)))
```

Comment:
nice so we've got backward test covered here

Reply:
honestly I wouldn't modify this test, there's no way it's going to be kept up to date as we add more complex losses.  Instead I would make sure we have both criterion and module tests that correspond to the existing tests.  It doesn't look like you have any criterion tests currently and you don't have a 1-to-1 correspondence with the existing modeul tests.